<!doctype html>
<html lang="">
<head>
    <meta charset="UTF-8">
    <meta name="viewport"
          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Test Results Report - Solar1</title>

    <link href="reportng.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<div id="meta">
Generated by <a href="http://www.testng.org" target="_blank">TestNG</a> with <a href="https://github.com/testng-team/reportng" target="_blank">ReportNG</a>
at 17:12 CDT on понедельник 03 апреля 2023
    <br/><span
    id="systemInfo">User@WIN-EV87Q1VQHF4&nbsp;/&nbsp;Java 19.0.1 (Oracle Corporation)&nbsp;/&nbsp;Windows 10 10.0 (amd64)</span>
</div>
<h1>Solar1</h1>

<p>
    <a href="index.html">Suites</a>
        &#183; <a href="output.html">Log Output</a>
</p>

<p>Test duration
    : 44,632s</p>


    <table class="resultsTable">
        <tr>
            <th colspan="3" class="header failed">Failed Tests</th>
        </tr>
            <tr>
                <td class="group" colspan="3">tests.TestCases</td>
            </tr>
    <tr>
        <td class="method">
                glitching_user_login
        </td>
        <td class="duration">
            7,285s
        </td>
        <td class="result">
            <!-- Display the dependencies for skipped test methods -->


            <!-- Show logger output for the test -->

                <a href="javascript:toggleElement('exception-0', 'block')"
                   title="Click to expand/collapse">
                    <b>org.openqa.selenium.remote.UnreachableBrowserException: Error communicating with the remote browser. It may have died.<br/>
Build info: version: &apos;4.4.0&apos;, revision: &apos;e5c75ed026a&apos;<br/>
System info: host: &apos;WIN-EV87Q1VQHF4&apos;, ip: &apos;192.169.108.3&apos;, os.name: &apos;Windows 10&apos;, os.arch: &apos;amd64&apos;, os.version: &apos;10.0&apos;, java.version: &apos;19.0.1&apos;<br/>
Driver info: org.openqa.selenium.chrome.ChromeDriver<br/>
Command: [0ffd588f5126cfebecafde225952c1f1, clickElement {id=391e245a-2909-4259-8765-b754974ec8b2}]<br/>
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 111.0.5563.147, chrome: {chromedriverVersion: 111.0.5563.64 (c710e93d5b63..., userDataDir: C:\Program Files (x86)\scop...}, goog:chromeOptions: {debuggerAddress: localhost:9679}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: WINDOWS, proxy: Proxy(), se:cdp: ws://localhost:9679/devtool..., se:cdpVersion: 111.0.5563.147, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}<br/>
Element: [[ChromeDriver: chrome on WINDOWS (0ffd588f5126cfebecafde225952c1f1)] -&gt; css selector: #login-button]<br/>
Session ID: 0ffd588f5126cfebecafde225952c1f1</b>
                </a>
                <br/>
                <div class="stackTrace" id="exception-0">
                        org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:571)<br/>
                        org.openqa.selenium.remote.RemoteWebElement.execute(RemoteWebElement.java:257)<br/>
                        org.openqa.selenium.remote.RemoteWebElement.click(RemoteWebElement.java:78)<br/>
                        pages.LoginPage.clickLoginBtn(LoginPage.java:24)<br/>
                        tests.TestCases.glitching_user_login(TestCases.java:59)<br/>
                        java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)<br/>
                        java.base/java.lang.reflect.Method.invoke(Method.java:578)<br/>
                        org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)<br/>
                        org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)<br/>
                        org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)<br/>
                        org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)<br/>
                        org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:962)<br/>
                        org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)<br/>
                        org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)<br/>
                        org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)<br/>
                        java.base/java.util.ArrayList.forEach(ArrayList.java:1511)<br/>
                        org.testng.TestRunner.privateRun(TestRunner.java:806)<br/>
                        org.testng.TestRunner.run(TestRunner.java:601)<br/>
                        org.testng.SuiteRunner.runTest(SuiteRunner.java:433)<br/>
                        org.testng.SuiteRunner.runSequentially(SuiteRunner.java:427)<br/>
                        org.testng.SuiteRunner.privateRun(SuiteRunner.java:387)<br/>
                        org.testng.SuiteRunner.run(SuiteRunner.java:330)<br/>
                        org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)<br/>
                        org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)<br/>
                        org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)<br/>
                        org.testng.TestNG.runSuitesLocally(TestNG.java:1176)<br/>
                        org.testng.TestNG.runSuites(TestNG.java:1099)<br/>
                        org.testng.TestNG.run(TestNG.java:1067)<br/>
                        com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)<br/>
                        com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:105)<br/>
                        <b>Caused by: </b>
                        <a href="javascript:toggleElement('exception-1', 'block')"
                           title="Click to expand/collapse">
                            java.io.UncheckedIOException: java.net.SocketException: Connection reset
                        </a>
                        <br/>
                        <div class="stackTrace" id="exception-1">
                                org.openqa.selenium.remote.http.netty.NettyHttpHandler.makeCall(NettyHttpHandler.java:73)<br/>
                                org.openqa.selenium.remote.http.AddSeleniumUserAgent.lambda$apply$0(AddSeleniumUserAgent.java:42)<br/>
                                org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:56)<br/>
                                org.openqa.selenium.remote.http.netty.NettyHttpHandler.execute(NettyHttpHandler.java:49)<br/>
                                org.openqa.selenium.remote.http.AddSeleniumUserAgent.lambda$apply$0(AddSeleniumUserAgent.java:42)<br/>
                                org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:56)<br/>
                                org.openqa.selenium.remote.http.netty.NettyClient.execute(NettyClient.java:98)<br/>
                                org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:181)<br/>
                                org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:167)<br/>
                                org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:142)<br/>
                                org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:547)<br/>
                                org.openqa.selenium.remote.RemoteWebElement.execute(RemoteWebElement.java:257)<br/>
                                org.openqa.selenium.remote.RemoteWebElement.click(RemoteWebElement.java:78)<br/>
                                pages.LoginPage.clickLoginBtn(LoginPage.java:24)<br/>
                                tests.TestCases.glitching_user_login(TestCases.java:59)<br/>
                                java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)<br/>
                                java.base/java.lang.reflect.Method.invoke(Method.java:578)<br/>
                                org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)<br/>
                                org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)<br/>
                                org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)<br/>
                                org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)<br/>
                                org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:962)<br/>
                                org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)<br/>
                                org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)<br/>
                                org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)<br/>
                                java.base/java.util.ArrayList.forEach(ArrayList.java:1511)<br/>
                                org.testng.TestRunner.privateRun(TestRunner.java:806)<br/>
                                org.testng.TestRunner.run(TestRunner.java:601)<br/>
                                org.testng.SuiteRunner.runTest(SuiteRunner.java:433)<br/>
                                org.testng.SuiteRunner.runSequentially(SuiteRunner.java:427)<br/>
                                org.testng.SuiteRunner.privateRun(SuiteRunner.java:387)<br/>
                                org.testng.SuiteRunner.run(SuiteRunner.java:330)<br/>
                                org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)<br/>
                                org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)<br/>
                                org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)<br/>
                                org.testng.TestNG.runSuitesLocally(TestNG.java:1176)<br/>
                                org.testng.TestNG.runSuites(TestNG.java:1099)<br/>
                                org.testng.TestNG.run(TestNG.java:1067)<br/>
                                com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)<br/>
                                com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:105)<br/>
                        </div>
                        <b>Caused by: </b>
                        <a href="javascript:toggleElement('exception-2', 'block')"
                           title="Click to expand/collapse">
                            java.net.SocketException: Connection reset
                        </a>
                        <br/>
                        <div class="stackTrace" id="exception-2">
                                java.base/sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:401)<br/>
                                java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:434)<br/>
                                io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:258)<br/>
                                io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1132)<br/>
                                io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:357)<br/>
                                io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151)<br/>
                                io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722)<br/>
                                io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658)<br/>
                                io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:584)<br/>
                                io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)<br/>
                                io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)<br/>
                                io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)<br/>
                                io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)<br/>
                                java.base/java.lang.Thread.run(Thread.java:1589)<br/>
                        </div>
                </div>
        </td>
    </tr>
    <tr>
        <td class="method">
                locked_out_user_login
        </td>
        <td class="duration">
            8,253s
        </td>
        <td class="result">
            <!-- Display the dependencies for skipped test methods -->


            <!-- Show logger output for the test -->

                <a href="javascript:toggleElement('exception-3', 'block')"
                   title="Click to expand/collapse">
                    <b>java.lang.AssertionError: User is locked out and should not be able to log in. expected [https://qa-challenge.codesubmit.io/inventory.html] but found [https://qa-challenge.codesubmit.io/]</b>
                </a>
                <br/>
                <div class="stackTrace" id="exception-3">
                        org.testng.Assert.fail(Assert.java:110)<br/>
                        org.testng.Assert.failNotEquals(Assert.java:1413)<br/>
                        org.testng.Assert.assertEqualsImpl(Assert.java:149)<br/>
                        org.testng.Assert.assertEquals(Assert.java:131)<br/>
                        org.testng.Assert.assertEquals(Assert.java:655)<br/>
                        tests.TestCases.locked_out_user_login(TestCases.java:38)<br/>
                        java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)<br/>
                        java.base/java.lang.reflect.Method.invoke(Method.java:578)<br/>
                        org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)<br/>
                        org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)<br/>
                        org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)<br/>
                        org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)<br/>
                        org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:962)<br/>
                        org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)<br/>
                        org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)<br/>
                        org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)<br/>
                        java.base/java.util.ArrayList.forEach(ArrayList.java:1511)<br/>
                        org.testng.TestRunner.privateRun(TestRunner.java:806)<br/>
                        org.testng.TestRunner.run(TestRunner.java:601)<br/>
                        org.testng.SuiteRunner.runTest(SuiteRunner.java:433)<br/>
                        org.testng.SuiteRunner.runSequentially(SuiteRunner.java:427)<br/>
                        org.testng.SuiteRunner.privateRun(SuiteRunner.java:387)<br/>
                        org.testng.SuiteRunner.run(SuiteRunner.java:330)<br/>
                        org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)<br/>
                        org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)<br/>
                        org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)<br/>
                        org.testng.TestNG.runSuitesLocally(TestNG.java:1176)<br/>
                        org.testng.TestNG.runSuites(TestNG.java:1099)<br/>
                        org.testng.TestNG.run(TestNG.java:1067)<br/>
                        com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)<br/>
                        com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:105)<br/>
                </div>
        </td>
    </tr>
    </table>


    <table class="resultsTable" width="100%">
        <tr>
            <th colspan="3" class="header passed">Passed Tests</th>
        </tr>
            <tr>
                <td class="group" colspan="3">tests.TestCases</td>
            </tr>
    <tr>
        <td class="method">
                problem_user_login
        </td>
        <td class="duration">
            14,329s
        </td>
        <td class="result">
            <!-- Display the dependencies for skipped test methods -->


            <!-- Show logger output for the test -->

        </td>
    </tr>
    <tr>
        <td class="method">
                standard_user_login
        </td>
        <td class="duration">
            9,349s
        </td>
        <td class="result">
            <!-- Display the dependencies for skipped test methods -->


            <!-- Show logger output for the test -->

        </td>
    </tr>
    </table>

<script type="text/javascript" src="reportng.js"></script>
</body>
